tests/trainer/test_trainer.py

```diff
@@ -2276,22 +2276,23 @@ def test_galore_lr_display_with_scheduler(self):
 
         # creating log history of trainer, results don't matter
         trainer.train()
-        logs = trainer.state.log_history[1:][:-1]
+        logs = trainer.state.log_history[1:-1]
 
         # reach given learning rate peak and end with 0 lr
-        self.assertTrue(logs[num_warmup_steps - 2]["learning_rate"] == learning_rate)
-        self.assertTrue(logs[-1]["learning_rate"] == 0)
+        self.assertTrue(logs[num_warmup_steps - 1]["learning_rate"] == learning_rate)
```

Comment:
For both A100 and XPU log(actually they are same), the `0.0002` `learning_rate` happens in step 6. Don't know T4's situation, but i suppose this is not planform-dependent behavior, so maybe case need to be updated. My `galore-torch`       version is 1.0

Reply:
Not sure I understand, it's the 6th item, so the index is `5`, which is `num_warmup_steps` ..?

Reply:
but maybe @SunMarc can check this part

Reply:
yes, the 6th item's index is 5 in 0-based indexing, and then original first item is sliced out, so the index becomes `num_warmup_steps-1`

Reply:
> but maybe @SunMarc can check this part

@SunMarc , could you shed your insights here? maybe in prior version of `galore-torch` the log is different from current version or other considerations. I am using `galore-torch`  1.0

Reply:
Our daily CI don't have `galore-torc` so we didn't see it on reports. Manually tested on T4, this PR indeed fixes the issue. So I will merge.

regarding if to add `galore-torch` in the corresponding CI docker images, I will wait @SunMarc response (here or internally).

Thank you @yao-matrix